### PR TITLE
Add Dropbox API V2 support on PHP5.5+

### PR DESCRIPTION
**Pull request by semgovaert**
My attempt for adding Dropbox API V2 support.

### CODE REVIEW

File: config/flysystem.php [hunk 1]
```diff
@@ -67,9 +67,8 @@
         ],
 
         'dropbox' => [
-            'driver'     => 'dropbox',
-            'token'      => 'your-token',
-            'app'        => 'your-app',
+            'driver'             => 'dropbox',
+            'authorizationToken' => 'your-authorization-token',
```
Comment (GrahamCampbell) [inline config/flysystem.php]: Please just call this `token`.

Comment (GrahamCampbell) [inline config/flysystem.php]: `'token' => 'your-token',`

File: composer.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
         "league/flysystem-aws-s3-v3": "AwsS3 adapter support.",
         "league/flysystem-azure": "Azure adapter support.",
         "league/flysystem-cached-adapter": "Adapter caching support.",
-        "league/flysystem-dropbox": "Dropbox adapter support.",
+        "srmklive/flysystem-dropbox-v2": "Dropbox adapter support.",
```
Comment (GrahamCampbell) [inline composer.json]: Is this the official package recommended by flysystem?

**Comment by GrahamCampbell** [comment]
Thanks for the PR. I've left some comments. Could you also update the readme too please?

File: src/Adapters/DropboxConnector.php [hunk 3]
```diff
@@ -28,7 +28,7 @@ class DropboxConnector implements ConnectorInterface
      *
      * @param string[] $config
      *
-     * @return \League\Flysystem\Dropbox\DropboxAdapter
+     * @return Srmklive\Dropbox\Adapter\DropboxAdapter
```
Comment (GrahamCampbell) [inline src/Adapters/DropboxConnector.php]: Missing leading slash.

File: src/Adapters/DropboxConnector.php [hunk 4]
```diff
@@ -11,10 +11,10 @@
 
 namespace GrahamCampbell\Flysystem\Adapters;
 
-use Dropbox\Client;
+use Srmklive\Dropbox\Client\DropboxClient as Client;
```
Comment (GrahamCampbell) [inline src/Adapters/DropboxConnector.php]: No need to alias this. :)

File: composer.json (hunk 2 above)
Comment (semgovaert) [inline composer.json]: Hi Graham,

Yes, league/flysystem-dropbox refers to spatie/flysystem-dropbox, which then refers to srmklive/flysystem-dropbox-v2 for PHP5 support. See: 
https://github.com/spatie/flysystem-dropbox/blob/master/README.md

File: config/flysystem.php (hunk 1 above)
Comment (semgovaert) [inline config/flysystem.php]: Done!

File: src/Adapters/DropboxConnector.php (hunk 3 above)
Comment (semgovaert) [inline src/Adapters/DropboxConnector.php]: Done!

File: src/Adapters/DropboxConnector.php (hunk 4 above)
Comment (semgovaert) [inline src/Adapters/DropboxConnector.php]: Done!

File: composer.json (hunk 2 above)
Comment (GrahamCampbell) [inline composer.json]: OK. I see. Thanks. 👍

File: src/Adapters/DropboxConnector.php [hunk 5]
```diff
@@ -50,23 +50,23 @@ public function connect(array $config)
      */
     protected function getAuth(array $config)
     {
-        if (!array_key_exists('token', $config) || !array_key_exists('app', $config)) {
-            throw new InvalidArgumentException('The dropbox connector requires authentication.');
+        if (!array_key_exists('token', $config)) {
+            throw new InvalidArgumentException('The dropbox connector requires authorization.');
```
Comment (GrahamCampbell) [inline src/Adapters/DropboxConnector.php]: Can we keep `authentication` please? Keeps it consistent across this package. :)

Comment (semgovaert) [inline src/Adapters/DropboxConnector.php]: my bad — done!

**Comment by GrahamCampbell** [comment]
Looks good. Thanks a lot. Will merge once travis completes. :)

**Comment by GrahamCampbell** [comment]
Just an update here, going to change this to the php 7 version.

**Comment by GrahamCampbell** [comment]
https://github.com/GrahamCampbell/Laravel-Flysystem/pull/97

**Pull request merged**